File: erl/lib/sundries/keypad.ex
defmodule Sundries.Keypad do
  def dial(wording) do
    wording
    |> String.upcase()
    |> String.replace(["A", "B", "C"], "2")
    |> String.replace(["D", "E", "F"], "3")
    |> String.replace(["G", "H", "I"], "4")
    |> String.replace(["J", "K", "L"], "5")
    |> String.replace(["M", "N", "O"], "6")
    |> String.replace(["P", "Q", "R", "S"], "7")
    |> String.replace(["T", "U", "V"], "8")
    |> String.replace(["W", "X", "Y", "Z"], "9")
    |> String.replace(" ", "0")
    |> String.replace(~r/[^0-9]/, "1")
  end
end

File: python-cli/sundries/keypad.py
import re
from functools import reduce

import click


@click.command()
@click.argument('wording')
def dial(wording: str):
    keypad = [
        ('ABC', '2'),
        ('DEF', '3'),
        ('GHI', '4'),
        ('JKL', '5'),
        ('MNO', '6'),
        ('PQRS', '7'),
        ('TUV', '8'),
        ('WXYZ', '9'),
        (' ', '0'),
        ('^0-9', '1'),
    ]

    click.echo(reduce(lambda s, k: re.sub('[{}]'.format(k[0]), k[1], s), keypad, wording.upper()))


if __name__ == '__main__':
    dial()
